pilot/pkg/security/authz/builder/builder.go

```diff
@@ -116,7 +128,7 @@ func build(policies []model.AuthorizationPolicy, tdBundle trustdomain.Bundle, fo
 				continue
 			}
 			m.MigrateTrustDomain(tdBundle)
-			generated, err := m.Generate(forTCP, forDeny)
+			generated, err := m.Generate(forTCP, forAllow)
```

Comment:
why do we need to switch from `forDeny` to `forAllow`?

Reply:
For audit I figured that rather than ignoring the whole rule when an http field is set for the tcp filter, we can just eliminate the offending field. This is the same behavior as the Deny action, making audit less restrictive. This could lead to a situation where more permissions and principals are auditted than desired, so it might be better to change it back.

Reply:
can you just pass the action type here? it looks cleaner considering we have 3 actions now.

Reply:
yes, I can do that.